dataplicity/directory_scanner.py

```diff
@@ -1,142 +1,83 @@
-from __future__ import print_function
-from __future__ import unicode_literals
+from __future__ import print_function, unicode_literals
 
+import json
 import logging
-import random
+import os.path
+import tempfile
+from threading import Lock, Thread
 from time import time
-from threading import Event, RLock, Thread
-from typing import Optional
+from typing import Callable, Optional
 
-from .scan_directory import scan_directory, ScanResult, ScanDirectoryError
-from . import jsonrpc
+from .compat import text_type
+from .scan_directory import ScanDirectoryError, ScanResult, scan_directory
 
 log = logging.getLogger("agent")
 
 
-class DirectoryScanner(Thread):
+class DirectoryScanner(object):
     """Periodically scans and uploads directory information."""
```

Comment:
I suppose the docstring would have to be altered as this is now an on-demand scan rather than perdiodical one, right ?

Reply:
Yep